contracts/CodexRecordCore.sol

```diff
@@ -52,19 +48,13 @@ contract CodexRecordCore is CodexRecordFees {
     internalMint(_to, newTokenId);
 
     // Add metadata to the newly created token
-    //
-    // @TODO: evaluate gas costs here, it may be more efficient to push each
-    //  individual file onto the existing fileHashes array for this index
-    //  instead of replacing the array altogether
     tokenData[newTokenId] = CodexRecordData({
       nameHash: _nameHash,
       descriptionHash: _descriptionHash,
       fileHashes: _fileHashes
     });
 
-    if (bytes(_providerId).length != 0 && bytes(_providerMetadataId).length != 0) {
-      emit Minted(newTokenId, _providerId, _providerMetadataId);
-    }
+    emit Minted(newTokenId, _data);
```

Comment:
I think this is important for the multi-provider story. Even though we won't have metadata about those other tokens we should still index their existence. In the future once we have a registry of providers, we could potentially even look up the metadata via an API call once this event comes in.

Reply:
The `Transfer` event is what indexes new Records (including third-party-created Records.) The `Minted` event was added because we couldn't emit the custom data with the `Transfer` event without becoming non-standards compliant. The `Minted` event is only useful is there's data to emit ¯\\\_(ツ)_/¯

(Not advocating reverting this or anything, just wanted to point out the futility)

Reply:
Ah that's a good point. Well I like your consistency point too so I'm going to leave this in there for now. I think it would be weird if some tokens fired a `Minted` event and others didn't, even though we can technically track that via other means.